android_app/app/src/main/java/com/health/openscale/core/bluetooth/BluetoothStandardWeightProfile.java

```diff
@@ -823,29 +825,59 @@ protected void chooseExistingScaleUser(Vector<ScaleUser> userList) {
         chooseScaleUserUi(choices);
     }
 
-    protected String getInitials(String fullName) {
-        if (fullName == null || fullName.isEmpty() || fullName.chars().allMatch(Character::isWhitespace)) {
-            return getDefaultInitials();
-        }
-        return buildInitialsStringFrom(fullName).toUpperCase();
+    protected String getInitials(final String fullName, final int size) {
+        if (0 == size)
+            return StringUtils.EMPTY;
+
+        if (StringUtils.isAnyBlank(fullName))
+            return getDefaultInitials(size);
+
+        return buildInitialsStringFrom(fullName, size).toUpperCase();
     }
 
-    private String getDefaultInitials() {
+    private String getDefaultInitials(final int size) {
+        if (1 > size)
+            return StringUtils.EMPTY;
+
         int userId = this.selectedUser.getId();
-        int userIndex = getUserScaleIndex(userId);
-        return "P" + userIndex + " ";
+        String userIndex = String.valueOf(getUserScaleIndex(userId));
+
+        StringBuilder stringBuilder = new StringBuilder();
+
+        if (userIndex.length() < size)
+            stringBuilder.append('P');
+
+        stringBuilder.append(userIndex);
+
+        final int padding = size - stringBuilder.length();
+
+        if(0 < padding)
+            stringBuilder.append(StringUtils.repeat(StringUtils.SPACE, padding));
```

Comment:
StringUtils.repeat() function accepts zero and negative values, it will generate zero-length string in such case;
```suggestion
        stringBuilder.append(StringUtils.repeat(StringUtils.SPACE, size - stringBuilder.length()));
```

Reply:
I have changed to logic a bit to be more clear what is going on there. The new approach is a good way to understand the code but also brings less calls, forward and back and memory allocation. So it will bring readability and speed benefits.

Reply:
My point is that the library function that you use `StringUtils.repeat()` - it does zero and negativity checks. This lib is specially tailored to reduce such checks in calling code.